packages/core/src/components/widgets/AltDateWidget.tsx

```diff
@@ -1,21 +1,27 @@
-import React, { Component } from "react";
-import PropTypes from "prop-types";
+import React, { Component, MouseEvent } from "react";
 
-import { shouldRender, parseDateString, toDateString, pad } from "@rjsf/utils";
+import {
+  shouldRender,
+  parseDateString,
+  toDateString,
+  pad,
+  WidgetProps,
+  DateObject,
+} from "@rjsf/utils";
 
-function rangeOptions(start, stop) {
-  let options = [];
+function rangeOptions(start: number, stop: number) {
+  const options = [];
   for (let i = start; i <= stop; i++) {
     options.push({ value: i, label: pad(i, 2) });
   }
   return options;
 }
 
-function readyForChange(state) {
+function readyForChange(state: any) {
   return Object.keys(state).every((key) => state[key] !== -1);
 }
 
-function DateElement(props) {
+function DateElement(props: any) {
```

Comment:
You can probably use `WidgetProps<T, F>` here as well.

Reply:
unfortunately not, since it also contains props from what `this.dateElementProps` returns.